web_api/core/test_views.py

```diff
@@ -173,6 +238,261 @@ def test_usage_billing_authentication(authed_client: Client, other_user: User) -
+    fake_subscription = stripe.Subscription.construct_from(
+        dict(
+            object="subscription",
+            id="sub_Gu1xedsfo1",
+            current_period_end=period_start,
+            current_period_start=period_end,
+            items=dict(data=[dict(object="subscription_item", id="si_Gx234091sd2")]),
+            plan=dict(id="plan_G2df31A4G5JzQ", object="plan", amount=499,),
+            quantity=4,
+            default_payment_method="pm_22dldxf3",
+        ),
+        "fake-key",
+    )
+
+    stripe_subscription_retrieve = mocker.patch(
+        "core.models.stripe.Subscription.retrieve", return_value=fake_subscription
+    )
+    stripe_subscription_modify = mocker.patch(
+        "core.models.stripe.Subscription.modify", return_value=fake_subscription
+    )
+    fake_invoice = stripe.Invoice.construct_from(
+        dict(object="invoice", id="in_00000000000000"), "fake-key",
+    )
+    stripe_invoice_create = mocker.patch(
+        "core.models.stripe.Invoice.create", return_value=fake_invoice
+    )
+    stripe_invoice_pay = mocker.patch("core.models.stripe.Invoice.pay")
+    account = Account.objects.create(
+        github_installation_id=377930,
+        github_account_id=900966,
+        github_account_login=user.github_login,
+        github_account_type="User",
+        stripe_customer_id="cus_Ged32s2xnx12",
+    )
+    AccountMembership.objects.create(account=account, user=user, role="admin")
+    StripeCustomerInformation.objects.create(
+        customer_id=account.stripe_customer_id,
+        subscription_id="sub_Gu1xedsfo1",
+        plan_id="plan_G2df31A4G5JzQ",
+        payment_method_id="pm_22dldxf3",
+        customer_email="accounting@acme-corp.com",
+        customer_balance=0,
+        customer_created=1585781308,
+        payment_method_card_brand="mastercard",
+        payment_method_card_exp_month="03",
+        payment_method_card_exp_year="32",
+        payment_method_card_last4="4242",
+        plan_amount=499,
+        subscription_quantity=3,
+        subscription_start_date=1585781784,
+        subscription_current_period_start=period_start,
+        subscription_current_period_end=period_end,
+    )
+    assert stripe_subscription_retrieve.call_count == 0
+    assert stripe_subscription_modify.call_count == 0
+    res = authed_client.post(
+        f"/v1/t/{account.id}/update_subscription",
+        dict(prorationTimestamp=period_start + 4 * ONE_DAY_SEC, seats=24),
+    )
+    assert res.status_code == 204
+    assert stripe_subscription_retrieve.call_count == 1
+    assert stripe_subscription_modify.call_count == 1
+    assert stripe_invoice_create.call_count == 1
+    assert stripe_invoice_pay.call_count == 1
+
+
+@pytest.mark.django_db
+def test_update_subscription_missing_customer(
+    authed_client: Client, user: User, other_user: User, mocker: Any
+) -> None:
```

Comment:
doc strings for all of these would make this behemoth more inviting

Reply:
also maybe we should move these tests into a more specific test file rather than `test_views`

something like `test_billing_views`

Reply:
not in this pr